src/Consumer.cc

```diff
@@ -247,16 +248,43 @@ class ConsumerReceiveWorker : public Napi::AsyncWorker {
 };
 
 Napi::Value Consumer::Receive(const Napi::CallbackInfo &info) {
-  Napi::Promise::Deferred deferred = Napi::Promise::Deferred::New(info.Env());
   if (info[0].IsUndefined()) {
+#if PULSAR_VERSION >= 3000000
```

Comment:
I don't think we need to handle the case of the pulsar CPP version before 3.0. For the next release, we will compile the node client based on the latest c++ client. If the user wants to compile the c++ client themselves, they should also use the latest c++ client. Otherwise, there will be a lot of redundant code for compatibility.

Reply:
+1

Reply:
I got it. I modified this PR to use the new C function directly.
PTAL